src/briefcase/integrations/subprocess.py

```diff
@@ -263,12 +342,13 @@ def cleanup(self, label, popen_process):
             to provide context in logging messages.
         :param popen_process: The Popen instance to clean up.
         """
-        popen_process.terminate()
-        try:
-            popen_process.wait(timeout=3)
-        except subprocess.TimeoutExpired:
-            self.command.logger.warning(f"Forcibly killing {label}...")
-            popen_process.kill()
+        with popen_process:
+            popen_process.terminate()
+            try:
+                popen_process.wait(timeout=3)
+            except subprocess.TimeoutExpired:
+                self.command.logger.warning(f"Forcibly killing {label}...")
+                popen_process.kill()
```

Comment:
Soo....`subprocess.Popen` interestingly does not expose a `close()` API...which would probably just be a wrapper on `__exit__()`. The consequence of not calling `__exit__()` is technically leads to leaking file/proxyfile handles and potentially creating zombie processes (although, as we know, these are cleaned up with Python exits). So, similar to ensuring we're calling `cleanup()` elsewhere, this ensures we also tell `Popen` to cleanup.

Since `Popen.__exit__()` is just calling `close()` on `stdout/err/in` and ensuring the process has exited, this should be safe to call multiple times therefore allowing callers to also open a `Popen` object via their own `with` without any issues. That is, as long as the expectation that calling `close()` multiple times, in general, in Python [remains safe](https://docs.python.org/3/library/io.html#io.IOBase.close).

Reply:
I actually ended up reverting this `with` block since closing the Popen process here while the thread may still be running can cause `readline` to throw an error (since `stdout` will have been closed). In general, not closing these resources certainly isn't the end of the world.....however, we may need to establish a more concrete lifecycle for them. Perhaps I'll look at this again in a future PR.